src/classes/ALLO_AllocationsUtil.cls

```diff
@@ -0,0 +1,1939 @@
+/*
```

Comment:
This class is way too big. It needs be split into discreet classes that perform one and only function for that class, even that turns into 4 different classes, each with tis own test class. There's definitely no benefit in putting all in a single class. If anything that will significantly complicate future maintenance of this. 

Plus, there are a lot of large static methods in the class that each have a lot of parameters. I'd prefer to see a single service class with a builder pattern that that has properties such as withAllocations(), withExclusions(), etc. and then individual instance methods to handle the processing of those records - probably with a lot more abstraction so that methods are much shorter and easier to follow.